src/pybrake/middleware/pyramid.py

```diff
@@ -0,0 +1,232 @@
+import logging
+import sys
+import time
+
+from .. import Notifier
+from .. import RouteMetric
+from ..metrics import (
+    set_active as set_active_metrics,
+    get_active as get_active_metrics,
+    start_span,
+    end_span,
+)
+
+logger = logging.getLogger(__name__)
+
+try:
+    from pyramid.httpexceptions import HTTPException
+    from pyramid.request import Request
+    from pyramid.threadlocal import get_current_request
+    from pyramid.config import Configurator
+    from pyramid import renderers, router
+except ImportError as e:
+    logger.error(str(e))
+    sys.exit()
+
+try:
+    from sqlalchemy import event
+    from pyramid_basemodel import Base
+except ImportError:
+    _sqla_available = False
+else:
+    _sqla_available = True
+
+try:
+    from pyramid_fullauth.request import request_user as current_user
+except ImportError:
+    _pyramid_login_available = False
+else:
+    _pyramid_login_available = True
+
+_UNKNOWN_ROUTE = "UNKNOWN"
+
+
+def request_filter(notice):
+    request = get_current_request()
+
+    if request is None:
+        return notice
+
+    ctx = notice["context"]
+    ctx["method"] = request.method
+    ctx["url"] = request.url
+    ctx["route"] = str(request.path)
+    ctx["userAddr"] = request.remote_addr
+
+    if _pyramid_login_available:
+        curr_user = current_user(request)
+        user = dict(id=curr_user.id)
+        for s in ["username", "firstname", "lastname"]:
+            if hasattr(curr_user, s):
+                user[s] = getattr(curr_user, s)
+        ctx["user"] = user
+
+    notice["params"]["Request"] = dict(
+        get=dict(request.GET),
+        post=dict(request.POST),
+        params=dict(request.params),
+        body=dict(request.body),
+        files=dict(request.body_file),
+        cookies=dict(request.cookies),
+        headers=dict(request.headers),
+        environ=request.environ,
+        view_args=request.urlargs,
+    )
+
+    return notice
+
+
+def handle_exception(notifier, exception):
+    notice = notifier.build_notice(exception)
```

Comment:
Why not just `notifier.send_notice(exception)`?

Reply:
`send_notice` in Pybrake accepts `dict` objects made by `build_notice` until now. We can supply an instance of `exception` directly to `send_notice`, but we must first update the `Notifier` class.

We should, in my opinion, initiate a new PR for that.

Reply:
Our README states we can pass the exception directly: https://github.com/airbrake/pybrake#sending-errors-to-airbrake 

Should that be updated?

Reply:
PTAL